Source/WebInspectorUI/UserInterface/Views/ConsoleMessageView.js

```diff
@@ -159,6 +162,37 @@ WI.ConsoleMessageView = class ConsoleMessageView extends WI.Object
         this._repeatCountElement.textContent = Number.abbreviate(count);
     }
 
+    _renderTimestamp()
+    {
+        if (!this._timestamp) {
+            if (this._timestampElement) {
+                this._timestampElement.remove();
+                this._timestampElement = null;
+            }
+            return;
+        }
+
+        if (!this._timestampElement) {
+            this._timestampElement = document.createElement("span");
+            this._timestampElement.classList.add("timestamp");
+            this._element.insertBefore(this._timestampElement, this._element.firstChild);
+        }
+
+        let date = new Date(this._timestamp * 1000);
+        var hours = date.getHours(),
+        minutes = date.getMinutes(),
+        seconds = date.getSeconds(),
+        milliseconds = date.getMilliseconds();
+
+        hours = (hours < 10) ? "0" + hours : hours;
+        minutes = (minutes < 10) ? "0" + minutes : minutes;
+        seconds = (seconds < 10) ? "0" + seconds : seconds;
+        milliseconds = (milliseconds < 100 && milliseconds >= 10) ? "0" + milliseconds : milliseconds;
+        milliseconds = (milliseconds < 10) ? "00" + milliseconds : milliseconds;
+
+        this._timestampElement.textContent = hours + ":" + minutes + ":" + seconds + ":" + milliseconds;
```

Comment:
Converting seconds to a human-readable timestamp would probably make a for a good utility method (similar to our existing `secondsToString` utility function).

Also, this isn't resilient to various localizations. For example in the US we would generally show the hours as a number from 1-12, with an AM/PM after the time. I think what we might actually want to do here is something along the lines of
```
let timeFormat = new Intl.DateTimeFormat("default", {hour: "2-digit", minute: "2-digit", second: "2-digit", fractionalSecondDigits: 3})

this._timestampElement.textContent = timeFormat.format(this._message.timestamp)
```
Which will follow locale-specific rules for showing the hours+minutes+seconds+milliseconds. We could probably make the time format a static member, like `WI.ConsoleMessageView.TimestampFormat` and reuse it for each message.

Reply:
could also maybe just use `Date.prototype.toLocaleString` 

also, we'd probably want `hour12: false` :)